### PR TITLE
feat(update.sh): [Don’t update kernel if kernel version is already installed]

### PR DESCRIPTION
**Pull request by Kreblskulm**
This will make it so that when the update is called it will check for the existence of 5.17.5 in the config files as an identifier for the kernel and if this identifier is found the kernel will not be installed due to it being installed already

### CODE REVIEW

**Comment by ajstrongdev** [comment]
Will review when I'm free. Until then I've assigned @wizard-28 to review.

File: update.sh [hunk 1]
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
...
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  sudo apt install ./*.deb
+  if [[ ! -f "$HOME/.rhino/config/5.17.5" ]]; then
+    cd ~/rhinoupdate/kernel/
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705_5.17.5-051705.202204271406_all.deb
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
+    sudo apt install ./*.deb
+    : > "$HOME/.rhino/config/5.17.5"
```
Comment (ajstrongdev) [inline update.sh]: The file should've been called 5-17-5. Minimal change but recommended.

File: update.sh [hunk 2]
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
 
 # If the user has selected the option to install the mainline kernel, install it onto the system.
 if [[ -f "$HOME/.rhino/config/mainline" ]]; then
-  cd ~/rhinoupdate/kernel/
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705_5.17.5-051705.202204271406_all.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  sudo apt install ./*.deb
+  if [[ ! -f "$HOME/.rhino/config/5.17.5" ]]; then
```
Comment (ajstrongdev) [inline update.sh]: File should be "5-17-5"

**Review by ajstrongdev — CHANGES_REQUESTED** [review]
Please change the file name to 5-17-5 rather than 5.17.5

**Review by ajstrongdev — APPROVED** [review]
All of the changes requested seem to have been fixed. Will test on Tuesday and then if it works as intended. It'll be merged.

File: update.sh [hunk 3]
```diff
@@ -61,5 +64,4 @@ fi
 
 # Allow the user to know that the upgrade has completed.
 echo "---
-The upgrade has been completed. Please reboot your system to see the changes.
----"
\ No newline at end of file
+The upgrade has been completed. Please reboot your system to see the changes.---"
```
Comment (wizard-28) [inline update.sh]: Why is this changed?

File: update.sh [hunk 4]
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
 
 # If the user has selected the option to install the mainline kernel, install it onto the system.
 if [[ -f "$HOME/.rhino/config/mainline" ]]; then
-  cd ~/rhinoupdate/kernel/
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-headers-5.17.5-051705_5.17.5-051705.202204271406_all.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  sudo apt install ./*.deb
+  if [[ ! -f "$HOME/.rhino/config/5-17-5" ]]; then
```
Comment (wizard-28) [inline update.sh:35]: Wouldn't it be better to check if the packages are already installed on the system by querying APT, instead of creating new files?

Comment (ajstrongdev) [reply update.sh:35]: I just believed this would be the most simple solution. Furthermore as Ubuntu devel upgrades their repositories there may be times where the Ubuntu kernel is installed over the mainline one, with matching version numbers. Wouldn't this create more issues in the long run?

File: update.sh (hunk 3 above)
Comment (ajstrongdev) [reply update.sh]: I am unsure. I'll review it in further detail when I am home.

Comment (ajstrongdev) [inline update.sh]: This needs to be fixed. It should be:
```sh
echo "---
The upgrade has been completed. Please reboot your system to see the changes
---"
```

**Review by ajstrongdev — CHANGES_REQUESTED** [review]
Minor change required on the last line which tells the user to reboot.

**Review by ajstrongdev — APPROVED** [review]
Seems to have fixed the mistake. Will merge after some testing.
